File: mobile/lib/data/models/customer.dart
class Customer {
  String userId;
  String firstName;
  String lastName;
  String birthDate;
  String cityOfBirth;
  String nicId;
  int phoneNumber;
  String email;
  String profilePicture;
  String password;
  String pin;

  Customer({
    required this.userId,
    required this.firstName,
    required this.lastName,
    required this.birthDate,
    required this.cityOfBirth,
    required this.nicId,
    required this.phoneNumber,
    required this.email,
    required this.profilePicture,
    required this.password,
    required this.pin,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      userId: json['userId'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      birthDate: json['birthDate'],
      cityOfBirth: json['cityOfBirth'],
      nicId: json['nicId'],
      phoneNumber: json['phoneNumber'],
      email: json['email'],
      profilePicture: json['profilePicture'],
      password: json['password'],
      pin: json['pin'],
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['firstName'] = firstName;
    data['lastName'] = lastName;
    data['birthDate'] = birthDate;
    data['cityOfBirth'] = cityOfBirth;
    data['nicId'] = nicId;
    data['phoneNumber'] = phoneNumber;
    data['email'] = email;
    data['profilePicture'] = profilePicture;
    data['password'] = password;
    data['pin'] = pin;
    return data;
  }
}

File: src/main/java/com/bank/core/customer/Customer.java
package com.bank.core.customer;

import jakarta.persistence.*;
import lombok.Builder;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDate;

import java.util.Objects;
@Builder
@Entity(name = "customers")
public class Customer {

    @Id
    private String customerId;
    private String firstName;
    private String lastName;
    private LocalDate birthDate;
    @Column(unique = true)
    private String nicId; // national identity card number
    private Boolean hasActiveAccount;
    private String pin;


    public Customer(String customerId, String firstName, String lastName, LocalDate birthDate,String nicId,Boolean hasActiveAccount) {
        this.customerId = customerId;
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.nicId = nicId;
        this.hasActiveAccount = hasActiveAccount;
    }

    public Customer() {
    }

    public String getCustomerId() {
        return customerId;
    }

    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public LocalDate getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate = birthDate;
    }



    public String getNicId() {
        return nicId;
    }

    public void setNicId(String nicId) {
        this.nicId = nicId;
    }


    public Boolean getHasActiveAccount() {
        return hasActiveAccount;
    }

    public void setHasActiveAccount(Boolean hasActiveAccount) {
        this.hasActiveAccount = hasActiveAccount;
    }


    public String getPin() {
        return pin;
    }

    public void setPin(String pin) {
        this.pin = pin;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Customer customer = (Customer) o;
        return customerId.equals(customer.customerId) && firstName.equals(customer.firstName) && lastName.equals(customer.lastName) && birthDate.equals(customer.birthDate) && cityOfBirth.equals(customer.cityOfBirth) && nicId.equals(customer.nicId) && phoneNumber.equals(customer.phoneNumber) && email.equals(customer.email) && profilePicture.equals(customer.profilePicture) && hasActiveAccount.equals(customer.hasActiveAccount) && password.equals(customer.password) && pin.equals(customer.pin);
    }

    @Override
    public int hashCode() {
        return Objects.hash(customerId, firstName, lastName, birthDate, nicId, hasActiveAccount, pin);
    }

    @Override
    public String toString() {
        return "Customer{" +
                "customerId='" + customerId + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", birthDate=" + birthDate +
                ", nicId='" + nicId + '\'' +
                ", hasActiveAccount=" + hasActiveAccount +
                ", pin='" + pin + '\'' +
                '}';
    }
}

